### DISCUSSION TITLE
Does fsmc_lcd only work with screens?

### DISCUSSION THREAD

**Discussion by ibxx0402** [stm32-rs/stm32f4xx-hal#809] [Q&A]
I have some questions regarding the fsmc_lcd, as I am unsure if it only works with screens or it's still able to connect to flash or ram. I am asking because it seems to be the recommended route for connecting external flash to a stm32 board.

**Comment by therealprof** [comment]
FSMC can work with other peripherals as well but those need a different setup and abstraction layer, that's why it is called fsmc_lcd to make the purpose of the implementation clear.

**Reply by ibxx0402** [reply]
Thanks for you quick answer.
Am I right in understanding that it is not recommended to try to use this implementation of FSMC to try to interface with external and flash or sram? And I would have to look at different options like spi or sdio.

**Reply by therealprof** [reply]
You're more than welcome to try to tweak the implementation to work with flash or sram (though you will have to find some interface other than `display-interface` to implement because that exclusively deals with displays) and you might even contribute that back to the project if you're successful. What I'm saying is: FSMC is non-trivial to use and currently only displays are supported by the existing implementation. If you like a good challenge, go for it. If you just want to get something to work quickly, you probably should look somewhere else or choose a different mechanism indeed.

**Reply by ibxx0402** [reply]
ok, thanks for the reply. I have done some more looking around found this it seems to be able to interface with sdram or Nand flash. It might also be able to work with other HAL's.

https://github.com/stm32-rs/stm32-fmc.
